CoreService/Items.psm1

```diff
@@ -156,10 +156,62 @@ Function Get-Item
 	}
 }
 
+function Test-Object
+{
+    <#
+    .Synopsis
+    Checks if the item with the given ID exists.
+	
+	.Notes
+    Example of properties available: Id, Title, etc.
+    
+    For a full list, consult the Content Manager Core Service API Reference Guide documentation 
+    (Tridion.ContentManager.Data.CommunicationManagement.IdentifiableObject object)
+	
+    .Inputs
+    None.
+
+    .Outputs
+    Returns a boolean type.
+
+    .Link
+    Get the latest version of this script from the following URL:
+    https://github.com/pkjaer/tridion-powershell-modules
+
+    .Example
+    Test-Object 'tcm:1-155-5110'
+    Returns if tcm tcm:1-155-5110 exists.
+
+    .Example
+    Test-Object '/webdav/02 Publication'
+    Returns if webdav path '/webdav/02 Publication' exists.
+    
+    #>
+    [CmdletBinding()]
+    Param
+    (
+		# The TCM URI of the user to load. If omitted, data for the current user is loaded instead.
+        [Parameter(ValueFromPipelineByPropertyName=$true)]
+		[ValidateNotNullOrEmpty()]
+        [string]$Id
+    )
+
+	try
```

Comment:
Thought about it, but still got an exception rising on tridion 2011 sp1.
Will check again later... Maybe I just wrote something wrong.
On Mar 21, 2016 11:18 AM, "Peter Kjaer" <notifications@github.com> wrote:

> In CoreService/Items.psm1
> <https://github.com/pkjaer/tridion-powershell-modules/pull/15#discussion_r56795293>
> :
>
> > +
> > +    .Example
> > +    Test-Object '/webdav/02 Publication'
> > +    Returns if webdav path '/webdav/02 Publication' exists.
> > +
> > +    #>
> > +    [CmdletBinding()]
> > +    Param
> > +    (
> > +		# The TCM URI of the user to load. If omitted, data for the current user is loaded instead.
> > +        [Parameter(ValueFromPipelineByPropertyName=$true)]
> > +		[ValidateNotNullOrEmpty()]
> > +        [string]$Id
> > +    )
> > +
> > +	try
>
> I would use $client.IsExistingObject instead of GetItem with a try/catch.
> No exceptions will be thrown and it also won't load the item as part of the
> test.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/pkjaer/tridion-powershell-modules/pull/15/files/6df18363718fe61a1c81cd454baa3ecb66312d51#r56795293>
>